src/test/scala/edu/ie3/mobsim/model/ChargingBehaviorSpec.scala

```diff
@@ -0,0 +1,374 @@
+/*
+ * © 2022. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.mobsim.model
+
+import edu.ie3.mobsim.io.geodata.PoiEnums.{
+  CategoricalLocationDictionary,
+  PoiTypeDictionary
+}
+import edu.ie3.mobsim.io.geodata.PoiEnums.PoiTypeDictionary.HOME
+import edu.ie3.test.common.UnitSpec
+import edu.ie3.util.quantities.PowerSystemUnits
+import tech.units.indriya.quantity.Quantities
+
+import java.time.ZonedDateTime
+import java.util.UUID
+
+class ChargingBehaviorSpec extends UnitSpec with ChargingBehaviorTestData {
+  "The ChargingBehavior" should {
+
+    "choose a chargingStation if charging is needed" in {
+      val uuid: Option[UUID] = ChargingBehavior.chooseChargingStation(
+        evChargingNeeded,
+        currentPricesAtChargingStations,
+        currentlyAvailableChargingPoints,
+        seed,
+        maxDistance
+      )
+
+      uuid.isDefined shouldBe true
+      uuid.map(_.toString) shouldBe Some("7537c0b6-3137-4e30-8a95-db1c0f9d9b81")
+    }
+
+    "choose no chargingStation if charging is not needed" in {
+      val uuid: Option[UUID] = ChargingBehavior.chooseChargingStation(
+        ev1.copyWith(ev1.getEStorage),
+        currentPricesAtChargingStations,
+        currentlyAvailableChargingPoints,
+        seed,
+        maxDistance
+      )
+
+      uuid.isDefined shouldBe false
+    }
+
+    "choose no chargingStation if no station is nearby" in {
+      val uuid: Option[UUID] = ChargingBehavior.chooseChargingStation(
+        evNoChargingStations,
+        currentPricesAtChargingStations,
+        noAvailableChargingPoints,
+        seed,
+        maxDistance
+      )
+
+      uuid.isDefined shouldBe false
+    }
+
+    "check if ev wants to charge" in {
+      val parkingTimeStart = ZonedDateTime.now()
+
+      val cases = Table(
+        (
+          "energy",
+          "destinationPoiTyp",
+          "destinationCategoricalLocation",
+          "destinationPoi",
+          "isChargingAtHomePossible",
+          "departureTime",
+          "seed",
+          "expectedResult"
+        ),
+        (
+          Quantities.getQuantity(39, PowerSystemUnits.KILOWATTHOUR),
+          PoiTypeDictionary.HOME,
+          CategoricalLocationDictionary.HOME,
+          poiHome,
+          true,
+          parkingTimeStart.plusMinutes(14),
+          seed,
+          false
+        ),
```

Comment:
Please briefly describe each scenario, so that one can comprehend which decisions in `chooseChargingStation` and `doesEvWantToCharge` are made.
So here, a description could be (for example): "EV with destination charging station that does not stay long enough for charging"